### PR TITLE
test(crr): add comprehensive unit test coverage for ContainerRecreateRequest controller

### LINKED ISSUE #2348: Add comprehensive unit tests covering CRR
## Description

The ContainerRecreateRequest (CRR) controller currently has no unit test coverage, despite containing complex reconciliation logic with multiple timeout scenarios, state transitions, and edge cases.

## Current State

```
pkg/controller/containerrecreaterequest/
├── crr_controller.go         (333 lines - 0% test coverage)
├── crr_event_handler.go      (97 lines - 0% test coverage)
└── NO TEST FILES             ❌
```

## Why This Matters

The CRR controller handles critical production scenarios:

1. **Timeout handling** - Response timeout (1min), active deadline, TTL cleanup
2. **Pod lifecycle coordination** - Handles pod deletion, UID mismatches during reconciliation
3. **Readiness gate management** - Controls traffic flow during container restarts
4. **Finalizer cleanup** - Prevents resource leaks
5. **Container status synchronization** - Tracks restart progress

## Proposed Solution

Add comprehensive unit tests covering

### PR DESCRIPTION
**Pull request by arnavgogia20**
### Ⅰ. Describe what this PR does

This PR adds comprehensive unit tests for the `ContainerRecreateRequest` (CRR) controller reconcile logic.

The tests validate existing behavior across critical lifecycle paths, including requeue handling, status synchronization, timeout processing, pod lifecycle edge cases, and finalizer-related logic.  
No production code or controller semantics are modified.

---

### Ⅱ. Does this pull request fix one issue?

fixes #2348

---

### Ⅲ. Describe how to verify it

Run the following command:

```bash
go test ./pkg/controller/containerrecreaterequest/... -v -cover
```

---

### Ⅳ. Special notes for reviews
- Tests assert current controller behavior only and do not redefine failure semantics
- No reliance on real time, time.Sleep, or timing-sensitive logic
- Requeue assertions avoid exact-duration checks to reduce brittleness
- Assertions are intentionally scoped to minimize maintenance overhead

### CODE REVIEW

**Comment by arnavgogia20** [comment]
Hii @FillZpp @furykerry , Can u pls consider reviewing the PR, thankyouu

**Comment by Kunal1522** [comment]
@arnavgogia20 there are some failing checks.look into them .i think it is flake but u can verify it on ur side as well

**Comment by arnavgogia20** [comment]
> @arnavgogia20 there are some failing checks.look into them .i think it is flake but u can verify it on ur side as well

@Kunal1522 @furykerry Thanks for the heads up, but I’m not seeing any failing checks on my end right now ......
only long-running E2E jobs that are still in progress.
I’ll keep an eye on them and re-run if any of them actually fail.

File: pkg/controller/containerrecreaterequest/crr_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+						},
+					},
+				},
+			},
+			existingObjs:       []client.Object{},
+			fakeClock:          clocktesting.NewFakeClock(time.Now()),
+			podReadinessGate:   true,
+			expectedPhase:      appsv1alpha1.ContainerRecreateRequestRecreating,
+			expectedRequeue:    0,
+			expectedFinalizers: []string{appsv1alpha1.ContainerRecreateRequestUnreadyAcquiredKey},
```
Comment (furykerry) [inline pkg/controller/containerrecreaterequest/crr_controller_test.go:417]: can u also check the crr status in the test?

File: pkg/util/imagejob/imagejob_reader.go [hunk 2]
```diff
@@ -204,14 +204,12 @@ func GetActiveJobsForNodeImage(reader client.Reader, nodeImage, oldNodeImage *ap
...
 			}
-			if !selector.Empty() {
-				if selector.Matches(labels.Set(nodeImage.Labels)) {
-					matched = true
-				}
-				if oldNodeImage != nil {
-					if selector.Matches(labels.Set(oldNodeImage.Labels)) {
-						oldMatched = true
-					}
+			if selector.Empty() || selector.Matches(labels.Set(nodeImage.Labels)) {
```
Comment (furykerry) [inline pkg/util/imagejob/imagejob_reader.go]: plz remove commit from unrelated patch

File: pkg/controller/containerrecreaterequest/crr_controller_test.go (hunk 1 above)
Comment (arnavgogia20) [reply pkg/controller/containerrecreaterequest/crr_controller_test.go:417]: @furykerry Yes i have made the changes accordingly...pls check thankyouu

File: pkg/util/imagejob/imagejob_reader.go (hunk 2 above)
Comment (arnavgogia20) [reply pkg/util/imagejob/imagejob_reader.go]: @furykerry Yes i have made the changes accordingly...pls check the PR now......thankyouu

**Comment by arnavgogia20** [comment]
Hii @furykerry and @FillZpp , the PR is review-ready finally, so requesting you to do it whenever u feel like....thankyouu
